commons/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java

```diff
@@ -269,6 +269,25 @@ public boolean equals(Object obj) {
       return true;
    }
 
+   public boolean matches(AttributeSet other) {
+      if (other.attributes.size() != attributes.size())
+         return false;
+      Iterator<Map.Entry<String, Attribute<?>>> i = attributes.entrySet().iterator();
+      while (i.hasNext()) {
+         Map.Entry<String, Attribute<?>> e = i.next();
+         String key = e.getKey();
+         Attribute<?> value = e.getValue();
+         if (value == null) {
+            if (!(other.attributes.get(key)==null && other.attributes.containsKey(key)))
```

Comment:
I would think we would want to check `containsKey` first as I would think it would be much more likely to have no entry than to have keys with null values.